### PR TITLE
Simplify read_stream and read_all API.

### PR DESCRIPTION
**Pull request by YoEight**
Changed: Simplify `read_stream` and `read_all` API.

The API is now closer to what other clients are doing.

### CODE REVIEW

**Comment by oskardudycz** [comment]
@YoEight, question, aren't those changes breaking? Looking at samples, it seems that the public API has changed.

**Comment by YoEight** [comment]
Yes it is and we have already merged many breaking changes but nothing has been released yet.

**Comment by oskardudycz** [comment]
@YoEight, I'm okay, just double-checking, as we shall merge it before the next major release 👍

File: src/client.rs [hunk 1]
```diff
@@ -88,40 +85,23 @@ impl Client {
...
-        let stream = commands::read_stream(
-            self.client.clone(),
-            options,
-            stream_name,
-            count.to_count() as u64,
-        )
-        .await?;
-
-        Ok(count.select(stream).await)
+        count: usize,
```
Comment (George-Payne) [inline src/client.rs:92]: The documentation site and most[^1] other clients name this option `maxCount`.  Also, they have it optional, with the default being reading all events in the stream. 

Is there a reason not to move it to `ReadStreamOptions` and have it part of it's default as `usize::MAX`?

[^1]: I only checked dotnet and node, so maybe not most.

Comment (oskardudycz) [reply src/client.rs:92]: Fair point 👍

Comment (YoEight) [reply src/client.rs]: Done

File: tests/integration.rs [hunk 2]
```diff
@@ -191,7 +191,7 @@ async fn test_tombstone_stream(client: &Client) -> Result<(), Box<dyn Error>> {
     debug!("Tombstone stream [{}] result: {:?}", stream_id, result);
 
     let result = client
-        .read_stream(stream_id.as_str(), &Default::default(), 1)
+        .read_stream(stream_id.as_str(), &Default::default())
```
Comment (George-Payne) [inline tests/integration.rs:194]: Shouldn't this be:

`ReadStreamOptions::default().max_count(1)` 

Or are there so few events in the stream it really doesnt matter?

**Review by George-Payne — APPROVED** [review]

**Review by oskardudycz — APPROVED** [review]

**Review by arwinneil — APPROVED** [review]

**Pull request merged**